src/library/scala/collection/IndexedSeq.scala

```diff
@@ -128,3 +131,22 @@ trait IndexedSeqOps[+A, +CC[_], +C] extends Any with SeqOps[A, CC, C] { self =>
     }
   }
 }
+
+/** A fast sliding iterator for IndexedSeqs which uses the underlying `slice` operation. */
+private final class IndexedSeqSlidingIterator[A, CC[_], C](s: IndexedSeqOps[A, CC, C], size: Int, step: Int)
+  extends Iterator[C] {
+
+  private[this] val len = s.length
+  private[this] var pos = 0
+
+  def hasNext: Boolean = pos < len
+
+  def next(): C = {
+    if(!hasNext) Iterator.empty.next()
+    val end = ((pos.toLong + size) min len).toInt
```

Comment:
Doesn't JS (at least in some flavors) have a speed penalty for using longs?  I would think code like

```
var end = pos + size
if (end < 0 || end > len) end = len
```
would be at least as fast on the JVM (maybe faster), and would be kinder to the JVM.  Then again, the slice probably takes most of the time, so maybe it's not important.

Reply:
JS indeed has a speed penalty when using Longs. Usually 3 to 5x slower than equivalent operations on ints.